### PR TITLE
DefaultMesh and STL update

### PR DESCRIPTION
**Pull request by kephale**
Moving imagej-ops DefaultMesh to this library and updating the STL
meshes to use it

### CODE REVIEW

**Comment by kephale** [comment]
I leave this PR open for debate. This contains @tibuch 's DefaultMesh from ops because we haven't discussed where this library fits in the dependency hierarchy. However, meshes made with this library *need* to be the same meshes that we're using in ops.

**Comment by kephale** [comment]
Pinging @rimadoma on this too

File: pom.xml [hunk 1]
```diff
@@ -3,9 +3,9 @@
 	<modelVersion>4.0.0</modelVersion>
 
 	<parent>
-		<groupId>org.scijava</groupId>
-		<artifactId>pom-scijava</artifactId>
-		<version>12.0.0</version>
+		<groupId>net.imagej</groupId>
+		<artifactId>pom-imagej</artifactId>
+		<version>16.5.0</version>
```
Comment (ctrueden) [inline pom.xml:8]: Please fix this... it needs to be `org.scijava:pom-scijava:12.0.0`, _not_ the obsolete `pom-imagej` anymore.

File: src/main/java/net/imagej/plugins/io/mesh/stl/BinarySTLFormat.java [hunk 2]
```diff
@@ -94,34 +96,34 @@
 	private static void writeFacet(final ByteBuffer buffer,
 		final STLFacet facet)
 	{
-		writeVector(buffer, facet.normal);
-		writeVector(buffer, facet.vertex0);
-		writeVector(buffer, facet.vertex1);
-		writeVector(buffer, facet.vertex2);
+		writeVector(buffer, facet.getNormal());
```
Comment (ctrueden) [inline src/main/java/net/imagej/plugins/io/mesh/stl/BinarySTLFormat.java:99]: This is great, but better to do as two separate commits. The first one adds @tibuch's mesh classes, and the second one changes the code here accordingly?

File: src/main/java/net/imagej/plugins/io/mesh/stl/BinarySTLFormat.java [hunk 3]
```diff
@@ -30,13 +30,15 @@
...
 
-import com.sun.javafx.geom.Vec3f;
+import net.imagej.plugins.io.mesh.Vertex;
 
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 import java.util.ArrayList;
 import java.util.List;
 
+import org.apache.commons.math3.geometry.euclidean.threed.Vector3D;
```
Comment (ctrueden) [inline src/main/java/net/imagej/plugins/io/mesh/stl/BinarySTLFormat.java:40]: Actually, is this change directly related to the change over to `Mesh`? Maybe this should be yet another separate commit? From JavaFX's `Vec3f` to Commons Math 3's `Vector3D`?

**Review by ctrueden — CHANGES_REQUESTED** [review]
Most important thing is to fix the POM parent. Other than that, would be nice to split the commits.

**Comment by kephale** [comment]
I opened a new pull request with the commits split: https://github.com/imagej/imagej-mesh/pull/5

**Comment by ctrueden** [comment]
@kephale wrote:
> I opened a new pull request with the commits split: #5

No need to do that. Just force push over the old branch, and the PR will magically update.

**Comment by ctrueden** [comment]
Closing in favor of #5.

**Pull request closed without merging**